nixos/modules/services/matrix/draupnir.nix

```diff
@@ -0,0 +1,250 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.draupnir;
+
+  yamlConfig = {
+    inherit (cfg) dataPath managementRoom protectedRooms verboseLogging;
+
+    accessToken = "@ACCESS_TOKEN@"; # will be replaced in "generateConfig"
+    homeserverUrl =
+      if cfg.pantalaimon.enable then
+        "http://${cfg.pantalaimon.options.listenAddress}:${toString cfg.pantalaimon.options.listenPort}"
+      else
+        cfg.homeserverUrl;
+
+    rawHomeserverUrl = cfg.homeserverUrl;
+
+    pantalaimon = {
+      inherit (cfg.pantalaimon) username;
+
+      use = cfg.pantalaimon.enable;
+      password = "@PANTALAIMON_PASSWORD@"; # will be replaced in "generateConfig"
+    };
+  };
+
+  moduleConfigFile = pkgs.writeText "module-config.yaml" (
+    generators.toYAML { } (filterAttrs (_: v: v != null)
+      (fold recursiveUpdate { } [ yamlConfig cfg.settings ])));
+
+  # these config files will be merged one after the other to build the final config
+  configFiles = [
+    "${pkgs.draupnir}/share/draupnir/config/default.yaml"
+    moduleConfigFile
+  ];
```

Comment:
Does draupnir not have default values, or why do we need to merge our config into the default config?

Reply:
It does have default values, don't do this please.